src/parser/parse_tree.h

```diff
@@ -764,11 +800,42 @@ struct json_t;
 #define PT_SYNONYM_OR_REPLACE(n)	((n)->info.synonym.or_replace)
 #define PT_SYNONYM_IF_EXISTS(n)		((n)->info.synonym.if_exists)
 
-#define PT_IS_SYNONYM_NODE(n) \
-	( (n)->node_type == PT_ALTER_SYNONYM || \
-	  (n)->node_type == PT_CREATE_SYNONYM || \
-	  (n)->node_type == PT_DROP_SYNONYM || \
-	  (n)->node_type == PT_RENAME_SYNONYM )
+/* Check node_type of PT_NODE */
+#define PT_NODE_IS_EXPR(n)		(PT_ASSERT_NOT_NULL ((n)), (n)->node_type == PT_EXPR)
+#define PT_NODE_IS_NAME(n)		(PT_ASSERT_NOT_NULL ((n)), (n)->node_type == PT_NAME)
+#define PT_NODE_IS_SPEC(n)		(PT_ASSERT_NOT_NULL ((n)), (n)->node_type == PT_SPEC)
+#define PT_NODE_IS_SYNONYM(n)		(PT_ASSERT_NOT_NULL ((n)), \
+					 (n)->node_type == PT_ALTER_SYNONYM || \
+					 (n)->node_type == PT_CREATE_SYNONYM || \
+					 (n)->node_type == PT_DROP_SYNONYM || \
+					 (n)->node_type == PT_RENAME_SYNONYM)
+
+/* PT_SPEC_INFO */
+#define PT_SPEC_GET_DB_OBJECT(n, r)	\
+	do \
+	  { \
+	    if (PT_SPEC_FLAT_ENTITY_LIST ((n)) != NULL \
+		&& PT_NAME_DB_OBJECT (PT_SPEC_FLAT_ENTITY_LIST ((n))) != NULL) \
+	      { \
+		(r) = PT_NAME_DB_OBJECT (PT_SPEC_FLAT_ENTITY_LIST ((n))); \
+	      } \
+	    if (PT_SPEC_ENTITY_NAME ((n)) != NULL) \
```

Comment:
Can't node->info.spec have flat_entity_listd and entity_name at the same time?
Is it intentional to use entity_name if all are present?

Reply:
I will fix it by changing it to "else if".